File: dotnet/Multiplayer/Rooms/Actions/ActionPollResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Rooms.Actions
{
    /// <summary>
    /// Response containing completed actions that were targeted to the polling player.
    /// </summary>
    /// <typeparam name="T">The type to deserialize action data into.</typeparam>
    public class ActionPollResponse<T> : ApiResponse<ERoomActionsPollError> where T : class, new()
    {
        /// <summary>
        /// List of completed actions with their details.
        /// </summary>
        public List<ActionInfo<T>> Actions { get; set; } = new();
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/actions/ActionPollResponse.java
package com.michitai.multiplayer.rooms.actions;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

import java.util.List;

/**
 * Response containing completed actions that were targeted to the polling player.
 *
 * @param <T> The type to deserialize action response data into.
 */
public class ActionPollResponse<T> extends ApiResponse {
    @JsonProperty("actions")
    private List<ActionInfo<T>> actions;

    public List<ActionInfo<T>> getActions() {
        return actions;
    }

    public void setActions(List<ActionInfo<T>> actions) {
        this.actions = actions;
    }
}
